runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -867,15 +902,15 @@ ClientSessionData::getClassRecord(J9Class *clazz, JITServer::ServerStream *strea
          );
          }
       }
-   else if (!record)
+   else if (uncachedClass)
       {
       // Request and cache class info from the client
       JITServerHelpers::ClassInfoTuple classInfoTuple;
       auto romClass = JITServerHelpers::getRemoteROMClass(clazz, stream, _persistentMemory, classInfoTuple);
       JITServerHelpers::cacheRemoteROMClassOrFreeIt(this, clazz, romClass, classInfoTuple);
 
       OMR::CriticalSection cs(getROMMapMonitor());
-      record = getClassRecord(clazz, missingLoaderRecord);
```

Comment:
Should pass down `missingLoaderInfo` - it can still be set to true. Can also `TR_ASSERT` that `uncachedClass` is false.